components/TextFields/src/MDCTextField.m

```diff
@@ -534,7 +530,7 @@ - (CGRect)textRectForBounds:(CGRect)bounds {
     // If there is a rightView, the clearButton will not be shown.
   } else {
     CGFloat clearButtonWidth = CGRectGetWidth(self.clearButton.bounds);
-    clearButtonWidth += 2 * MDCTextInputClearButtonImageBuiltInPadding;
+    clearButtonWidth += 2 * (MDCTextInputClearButtonImageBuiltInPadding * -1);
```

Comment:
nit: please stay consistent with your `* -1` placement here and below

Reply:
Oh no... It seems pretty broken with Hebrew, for example. Even on develop. Here's Hebrew on develop:

![simulator screen shot - iphone x - 2018-09-14 at 11 31 45](https://user-images.githubusercontent.com/8020010/45559990-ed4cfa80-b811-11e8-98d7-4a10a82f07e0.png)

That's as far left the cursor goes :(

Reply:
how does it look like in RTL though? :) this is the LTR layout right?

Reply:
also, yeah, this is a problem. probably should be fixed in a follow up

Reply:
Same deal.. Messed up:
![simulator screen shot - iphone x - 2018-09-14 at 11 37 01](https://user-images.githubusercontent.com/8020010/45560225-854ae400-b812-11e8-9b41-87c568ddcc7f.png)
FWIW I know @willlarche has verified with an Apple engineer at WWDC that there are bugs with cursor location/textRect on UITextField in certain RTL situations.

Reply:
It's true. We could do a workaround tho where the text insets are enlarged laterally.

Reply:
So @yarneo, there are still clearly issues with RTL in text fields in general, but what do you think about merging this as a step in the right direction? One that also happens to address the immediate problem described in the issue.

Reply:
sgtm